Spatial queries tutorial Closes #263

fax4ever opened the pull request:
Closes #263 

To test it you need to create a snapshot build from the contribution https://github.com/infinispan/infinispan/pull/13045.

karesti commented on infinispan-remote/spatial-queries/pom.xml:
```diff
@@ -0,0 +1,93 @@
+            <artifactId>infinispan-client-hotrod</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.infinispan</groupId>
```
do we still need to add all these dependencies ?

karesti reviewed (changes requested):
The only missing change is the documentation ! 
Nice job!!

karesti commented:
@fax4ever we are not using ISPN JIRA anymore for this project. A Github issue already exists for this tutorial:

https://github.com/infinispan/infinispan-simple-tutorials/issues/263

Please update the message of the PR too and add Closes #263 in the commit message please

karesti commented:
The build  should work with the latest snapshot version if using docker or podman locally

karesti commented:
@fax4ever there is an issue in the pom

```xml
Error: ] Some problems were encountered while processing the POMs:
Error:  Non-resolvable import POM: The following artifacts could not be resolved: org.infinispan:infinispan-bom:pom:15.1.0-SNAPSHOT (absent): Could not find artifact org.infinispan:infinispan-bom:pom:15.1.0-SNAPSHOT @ line 17, column 25
Warning:  The expression ${version} is deprecated. Please use ${project.version} instead. @ 
 @ 
Error:  The build could not read 1 project -> [Help 1]
Error:    
Error:    The project org.infinispan.tutorial.simple:infinispan-simple-tutorials-spatial-queries:1.0.0-SNAPSHOT (/home/runner/work/infinispan-simple-tutorials/infinispan-simple-tutorials/infinispan-remote/spatial-queries/pom.xml) has 1 error
Error:      Non-resolvable import POM: The following artifacts could not be resolved: org.infinispan:infinispan-bom:pom:15.1.0-SNAPSHOT (absent): Could not find artifact org.infinispan:infinispan-bom:pom:15.1.0-SNAPSHOT @ line 17, column 25 -> [Help 2]
Error:  
Error:  To see the full stack trace of the errors, re-run Maven with the -e switch.
Error:  Re-run Maven using the -X switch to enable full debug logging.
Error:  
Error:  For more information about the errors and possible solutions, please read the following articles:
Error:  [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/ProjectBuildingException
Error:  [Help 2] http://cwiki.apache.org/confluence/display/MAVEN/UnresolvableModelException
```

karesti commented:
@fax4ever rebase. target development branch

fax4ever commented:
Hi @karesti. I rebased getting an error that is the same if run a build from the base [infinispan:development](https://github.com/infinispan/infinispan-simple-tutorials/tree/development) branch. So I'm not really sure it is related this contribution.

This is the error:
```
(base) fax@fax-workstation:~/code/infinispan-simple-tutorials$ git checkout development 
Switched to branch 'development'
Your branch is up to date with 'origin/development'.
(base) fax@fax-workstation:~/code/infinispan-simple-tutorials$ mvn clean install -pl infinispan-remote/query/
[INFO] Scanning for projects...
[ERROR] [ERROR] Some problems were encountered while processing the POMs:
[ERROR] 'dependencies.dependency.version' for org.infinispan:infinispan-hotrod:jar is missing. @ line 45, column 21
[WARNING] The expression ${version} is deprecated. Please use ${project.version} instead. @ 
 @ 
[ERROR] The build could not read 1 project -> [Help 1]
[ERROR]   
[ERROR]   The project org.infinispan.tutorial.simple:infinispan-simple-tutorials-reactive:1.0.0-SNAPSHOT (/home/fax/code/infinispan-simple-tutorials/infinispan-remote/reactive-api/pom.xml) has 1 error
[ERROR]     'dependencies.dependency.version' for org.infinispan:infinispan-hotrod:jar is missing. @ line 45, column 21
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/ProjectBuildingException
```

karesti commented:
@fax4ever you're right, developement branch is broken
I'll fix it. I don't want to merge this before that

karesti commented:
@fax4ever you also need to update the docs

fax4ever commented:
Thanks @karesti
